src/aws.js

```diff
@@ -1,41 +1,54 @@
-      'export INSTANCE_ID=$(cat /var/lib/cloud/data/instance-id)',
-      `./config.sh --url https://github.com/${config.githubContext.owner}/${config.githubContext.repo} --token ${githubRegistrationToken} --name "$INSTANCE_ID" --labels ${label}`,
-      './run.sh',
-    ];
+    userdata_prefix.push(`cd "${config.input.runnerHomeDir}"`);
   } else {
-    return [
-      '#!/bin/bash',
-      'mkdir actions-runner && cd actions-runner',
-      'case $(uname -m) in aarch64) ARCH="arm64" ;; amd64|x86_64) ARCH="x64" ;; esac && export RUNNER_ARCH=${ARCH}',
+    userdata_prefix.push(
+      'mkdir -p actions-runner && cd actions-runner',
+      'case $(uname -m) in aarch64) RUNNER_ARCH="arm64" ;; amd64|x86_64) RUNNER_ARCH="x64" ;; esac && export RUNNER_ARCH',
       'curl -O -L https://github.com/actions/runner/releases/download/v2.280.3/actions-runner-linux-${RUNNER_ARCH}-2.280.3.tar.gz',
       'tar xzf ./actions-runner-linux-${RUNNER_ARCH}-2.280.3.tar.gz',
-      'export RUNNER_ALLOW_RUNASROOT=1',
-      'export DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=1',
-      'export INSTANCE_ID=$(cat /var/lib/cloud/data/instance-id)',
-      `./config.sh --url https://github.com/${config.githubContext.owner}/${config.githubContext.repo} --token ${githubRegistrationToken} --name "$INSTANCE_ID" --labels ${label}`,
-      './run.sh',
-    ];
+    );
   }
+  // push returns new size of array, so don't use its result as the function return value
+  return userdata_prefix.concat(
+    'INSTANCE_ID=$(cat /var/lib/cloud/data/instance-id)',
+    'TOKEN_HEADER="X-aws-ec2-metadata-token: $(curl -sX PUT "http://169.254.169.254/latest/api/token" -H "X-aws-ec2-metadata-token-ttl-seconds: 21600")"',
+    'AWS_DEFAULT_REGION="$(curl -sH "$TOKEN_HEADER" http://169.254.169.254/latest/meta-data/placement/availability-zone | sed -e"s/[a-z]*\$//")"',
+    'export AWS_DEFAULT_REGION',
+    // yikes
+    // avoid `aws ec2 ... | while read`, since the pipeline starts a subshell for the `while`
+    // so variables set in the loop disappear when it's done
+    'while read -r key value; do declare "RUNNER_$(echo -n "$key" | tr \'[:lower:]\' \'[:upper:]\' | tr -cs \'[:alnum:]\' _ )"="$value"; done < <(aws ec2 describe-tags --output text --filter Name=resource-id,Values="$INSTANCE_ID" --query "Tags[?starts_with(Key, \\`Ec2GithubRunner:\\`)].[Key, Value]" --output text | cut -f2- -d:)',
+    'RUNNER_ALLOW_RUNASROOT=1',
+    'DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=1',
+    'export RUNNER_ALLOW_RUNASROOT DOTNET_SYSTEM_GLOBALIZATION_INVARIANT',
+    `./config.sh --url "$RUNNER_URL" --token ${githubRegistrationToken} --name "$INSTANCE_ID" --labels "$RUNNER_LABEL"`,
+    './run.sh',
+  );
+}
+
+function makeMetadataTags(label) {
+    const { owner, repo } = github.context.repo;
+    const url = `${github.context.serverUrl}/${owner}/${repo}`
+    return [ { Key: "Ec2GithubRunner:Url", Value: url }, { Key: "Ec2GithubRunner:Label", Value: label }, ];
```

Comment:
There are some generic guidelines for tagging in notion here fwiw https://www.notion.so/freeagent/AWS-Tagging-Policy-a752ee992cbe4e49af97a417f93de750

TL;DR it's best to opt for lower-case to avoid typos and tags not been found down the line because someone got the casing incorrect. Hyphens for word separation and any custom ones specifc to fa prefixed with freeagent: to avoid name space collision issues on tags.